### PR TITLE
[11.0][ADD] website_sale_attribute_filter_category: New module

### PR DESCRIPTION
**Pull request by sergio-teruel**
Allow to display attributes filter grouped by category.
cc @Tecnativa

### CODE REVIEW

File: website_sale_attribute_filter_category/README.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/e-commerce-11-0/e-commerce-11-0-website_sale_attribute_filter_category
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/113/11.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+This module extends the functionality of website sale module to allow group
```
Comment (pedrobaeza) [inline website_sale_attribute_filter_category/README.rst]: s/to allow group/to allow to

File: website_sale_attribute_filter_category/README.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+To use this module you need to:
+
+* After the module has been installed.
```
Comment (pedrobaeza) [inline website_sale_attribute_filter_category/README.rst]: Remove this useless line

File: website_sale_attribute_filter_category/README.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+* Go to Website Shop and you can see product attributes filter grouped by
```
Comment (pedrobaeza) [inline website_sale_attribute_filter_category/README.rst]: Split this line as it contains several steps. Number the steps with `#.`

File: website_sale_attribute_filter_category/README.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+Authors
+~~~~~~~
+
+* Tecnativa
+
+Contributors
+~~~~~~~~~~~~
+
+* Sergio Teruel <sergio.teruel@tecnativa.com>
```
Comment (pedrobaeza) [inline website_sale_attribute_filter_category/README.rst]: Use "long" form with Tecnativa and URL in first level and your name in second (I recommend you to not put email, but as your wish).

File: website_sale_attribute_filter_category/controllers/main.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2019 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo.addons.website_sale.controllers.main import WebsiteSale
```
Comment (pedrobaeza) [inline website_sale_attribute_filter_category/controllers/main.py:3]: Is this compatible with other modules overwriting same class?

File: website_sale_attribute_filter_category/hooks.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2019 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import SUPERUSER_ID
+from odoo.api import Environment
+
+
+def post_init_hook(cr, registry):
+    """ Disable default Product Attribute Filter View """
+    env = Environment(cr, SUPERUSER_ID, {})
+    env.ref('website_sale.products_attributes').active = False
```
Comment (pedrobaeza) [inline website_sale_attribute_filter_category/hooks.py]: This is not legit. What happen with views inheriting this one? You have to add a children view to this one with your changes and with priority="999"

File: website_sale_attribute_filter_category/static/src/js/website_sale_attribute_filter_category.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+odoo.define('website_sale_attribute_filter_category.animation', function (require) {
+'use strict';
+
+var core = require('web.core');
+var sAnimation = require('website.content.snippets.animation');
+
+
+sAnimation.registry.sale_attributes_filter_category = sAnimation.Class.extend({
```
Comment (pedrobaeza) [inline website_sale_attribute_filter_category/static/src/js/website_sale_attribute_filter_category.js]: Why is this needed? Isn't there a class inside Bootstrap for collapsable content without JS?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: website_sale_attribute_filter_category/controllers/main.py (hunk 5 above)
Comment (sergio-teruel) [reply website_sale_attribute_filter_category/controllers/main.py:3]: This is a controller, I only add data to qcontext.

File: website_sale_attribute_filter_category/static/src/js/website_sale_attribute_filter_category.js (hunk 7 above)
Comment (sergio-teruel) [reply website_sale_attribute_filter_category/static/src/js/website_sale_attribute_filter_category.js]: Is only for icon >

File: website_sale_attribute_filter_category/controllers/main.py (hunk 5 above)
Comment (pedrobaeza) [reply website_sale_attribute_filter_category/controllers/main.py:3]: Yeah, mine is a general question about controllers. In my understanding, as this uses Python inheritance, you won't have all the inheritance chain through this. Am I right?

File: website_sale_attribute_filter_category/static/src/js/website_sale_attribute_filter_category.js (hunk 7 above)
Comment (pedrobaeza) [reply website_sale_attribute_filter_category/static/src/js/website_sale_attribute_filter_category.js]: Is this the way is done in Collapsable product categories?

Comment (pedrobaeza) [reply website_sale_attribute_filter_category/static/src/js/website_sale_attribute_filter_category.js]: If needed, please put a comment about what this does

File: website_sale_attribute_filter_category/hooks.py (hunk 6 above)
Comment (sergio-teruel) [reply website_sale_attribute_filter_category/hooks.py]: Inherit this view is complicated... but i will try to do it

File: website_sale_attribute_filter_category/controllers/main.py (hunk 5 above)
Comment (sergio-teruel) [reply website_sale_attribute_filter_category/controllers/main.py:3]: Yes

File: website_sale_attribute_filter_category/static/src/js/website_sale_attribute_filter_category.js (hunk 7 above)
Comment (sergio-teruel) [reply website_sale_attribute_filter_category/static/src/js/website_sale_attribute_filter_category.js]: https://github.com/odoo/odoo/blob/2156af480fddc09b5897b349c04c4c9a2d67c802/addons/website_sale/static/src/js/website_sale.js#L58

Comment (sergio-teruel) [reply website_sale_attribute_filter_category/static/src/js/website_sale_attribute_filter_category.js]: I did not see these lines. Thanks
I think that with bootstrap4 is not necessary

**Comment by chienandalu** [comment]
I'm trying to rebuild runbot but it's having such a short lifespan :disappointed:

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done and remove javascript code, replaced by less

File: website_sale_attribute_filter_category/README.rst (hunk 1 above)
Comment (pedrobaeza) [reply website_sale_attribute_filter_category/README.rst]: This has to be attended. I guess you have taken an outdated base, as I'm seeing other changes you made undone.

File: website_sale_attribute_filter_category/__manifest__.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    'application': False,
+    'installable': True,
+    'depends': [
+        'website_sale_comparison',
+    ],
+    'data': [
+        'views/assets.xml',
+        'views/templates.xml',
+    ],
+    'post_init_hook': 'post_init_hook',
```
Comment (pedrobaeza) [inline website_sale_attribute_filter_category/__manifest__.py]: Remove this

File: website_sale_attribute_filter_category/views/templates.xml [hunk 9]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 Tecnativa - Sergio Teruel
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+
+    <template id="products_attributes" inherit_id="website_sale.products_attributes">
```
Comment (pedrobaeza) [inline website_sale_attribute_filter_category/views/templates.xml]: priority="9999"

**Review by rafaelbn — APPROVED** [review]
Tested functionally :+1: 

@sergio-teruel review travis :red_circle:  please

**Comment by pedrobaeza** [comment]
Travis is red

**Comment by yajo** [comment]
It seems it cannot find this selector: https://travis-ci.org/OCA/e-commerce/jobs/517671229#L2204-L2205

**Comment by pedrobaeza** [comment]
Again red...

File: website_sale_attribute_filter_category/static/src/less/website_sale_attribute_filter_category.less [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+[data-toggle="collapse"] .fas:before {
+  content: "\f105";
+}
+
+[aria-expanded="true"] .fas:before {
+  content: "\f107";
+}
+
+[data-toggle="collapse"].collapsed .fas:before {
+  content: "\f105";
```
Comment (yajo) [inline website_sale_attribute_filter_category/static/src/less/website_sale_attribute_filter_category.less:15]: I'd appreciate some comments on this source code, because reading this isn't obvious what it does. I guess it's adding FA icons, but don't know which anyways.

File: website_sale_attribute_filter_category/views/templates.xml [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+<odoo>
+
+    <template id="products_attributes" inherit_id="website_sale.products_attributes" priority="9999">
+        <xpath expr="//form/ul" position="replace">
+                <h4>
+                    <i class="fa fa-filter" aria-hidden="true"/> <span>Filters by category</span>
+                </h4>
+                <t t-foreach="attribute_categories" t-as="attr_category">
+                    <t t-set="attributes_filtered" t-value="attributes.filtered(lambda x: x.category_id.id == attr_category[0])"/>
+                    <t t-set="display_category" t-value="any(len(a.value_ids) &gt;1 for a in attributes_filtered)"/>
```
Comment (yajo) [inline website_sale_attribute_filter_category/views/templates.xml:13]: IMHO this logic would be better handled in the controller, so the view gets only the data it should display.

File: website_sale_attribute_filter_category/views/templates.xml [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+                        <ul class="nav nav-stacked o_shop_collapse_attribute_filter" id="accordion">
+                            <li>
+                                <t t-set="values_selected" t-value="bool(attrib_set &amp; set(attributes_filtered.mapped('value_ids').ids))"/>
+                                <a data-toggle="collapse" t-attf-href="#collapse{{attr_category_index}}">
+                                    <i class="fa fas"/>
+                                    <span t-esc="attr_category[1]"/>
+                                </a>
+                                <ul t-attf-id="collapse{{attr_category_index}}" t-attf-class="nav nav-pills nav-stacked #{'collapse in' if values_selected else 'collapse'}">
+
+                                    <t>$0</t>
```
Comment (yajo) [inline website_sale_attribute_filter_category/views/templates.xml:24]: Did you mean this?

```suggestion
                                    <t t-raw="0" />
```

**Review by yajo — COMMENTED** [review]
I see that the HTML that appears in Travis logs doesn't contain any `.js_attributes` element, so I understand why the tour is failing.

I don't see that in your code, so I don't know where should it come from, but maybe some comment below can help you...

File: website_sale_attribute_filter_category/views/templates.xml (hunk 12 above)
Comment (sergio-teruel) [reply website_sale_attribute_filter_category/views/templates.xml:24]: No

File: website_sale_attribute_filter_category/tests/test_website_sale_attribute_filter_category.py [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+
+
+class websiteSaleAttributeFilterCategoryHttpCase(HttpCase):
+
+    def setUp(self):
+        super(websiteSaleAttributeFilterCategoryHttpCase, self).setUp()
+        # Activate the Product Attribute's Filter for testing environment
+        with self.cursor() as cr:
+            env = self.env(cr)
+            env.ref('website_sale.products_attributes').active = True
```
Comment (yajo) [inline website_sale_attribute_filter_category/tests/test_website_sale_attribute_filter_category.py]: This is no longer supported thanks to https://github.com/odoo/odoo/pull/30768. You have to choose between:

1. Activate the view in the tour by clicking on "Customize". You're running the tour as admin, so it should be OK.
2. Activate the view in demo data, which is loaded before this test starts.

**Review by yajo — CHANGES_REQUESTED** [review]

**Comment by sergio-teruel** [comment]
Squased!!

**Review by yajo — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
My comments about README are still unattended + several from @Yajo

File: website_sale_attribute_filter_category/static/src/less/website_sale_attribute_filter_category.less (hunk 10 above)
Comment (sergio-teruel) [reply website_sale_attribute_filter_category/static/src/less/website_sale_attribute_filter_category.less:15]: Done

**Review by pedrobaeza — APPROVED** [review]
